src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunMetadataFieldLayoutAlgorithm.cs

```diff
@@ -803,10 +803,17 @@ protected override ComputedInstanceFieldLayout ComputeInstanceFieldLayout(Metada
                 return ComputeExplicitFieldLayout(type, numInstanceFields);
             }
             else
+#if FEATURE_SEQUENTIAL_LAYOUT_WITH_REFS
+            if (ShouldLayoutSequential(type, out var isSequentialWithRefs))
+            {
+                return ComputeSequentialFieldLayout(type, numInstanceFields, isSequentialWithRefs);
+            }
+#else
             if (type.IsEnum || MarshalUtils.IsBlittableType(type) || IsManagedSequentialType(type))
```

Comment:
Note that this logic got a major overhaul in https://github.com/dotnet/runtime/pull/61759 .  I think you will want to do matching cleanup of `FEATURE_SEQUENTIAL_LAYOUT_WITH_REFS` mode once you pick up that change. I hope that it will make things simpler overall.

Reply:
Thanks for letting us know about this pull request, it looks very useful!